cli/azd/pkg/tools/tool.go

```diff
@@ -31,3 +36,38 @@ func toolInPath(name string) (bool, error) {
 		return false, fmt.Errorf("failed searching for `%s` on PATH: %w", name, err)
 	}
 }
+
+type ErrSemver struct {
+	ToolName    string
+	ToolRequire ToolMetaData
```

Comment:
Given how this is used now, I think we should rename this member to `UpgradeMessage`

Reply:
Sorry, I retract this statement - I didn't completely read what you had here and made a bad assumption as to how this worked.  What you ended up doing is good.